nlp/2021_06_29_long_range_transformers/LongRangeTransformers.ipynb

```diff
@@ -0,0 +1,1940 @@
+    "    tokenizer =  RobertaTokenizer.from_pretrained('roberta-base')\n",
+    "    if tokencount_type == IMDBDatasetTokenCountType.LONG:\n",
+    "        return ds.filter(lambda x: len(tokenizer.batch_encode_plus([x['text']]).input_ids[0]) > 512)\n",
+    "    if tokencount_type == IMDBDatasetTokenCountType.SHORT:\n",
+    "        return ds.filter(lambda x: len(tokenizer.batch_encode_plus([x['text']]).input_ids[0]) < 512).filter(lambda x: len(tokenizer.batch_encode_plus([x['text']]).input_ids[0]) > 256)\n",
+    "    \n",
+    "def filter_train_test_dataset_with_tokencount(train_ds, test_ds, tokencount_type):\n",
+    "    filt_train_ds =  filter_dataset_with_tokencount(train_ds, tokencount_type)\n",
+    "    filt_test_ds = filter_dataset_with_tokencount(test_ds, tokencount_type)\n",
+    "    return filt_train_ds, filt_test_ds"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 💥 Models\n",
+    "\n",
+    "Run the cells below to redo the training of each model\n",
+    "\n",
+    "🛎️ Disclaimer: this will take some time, you could also opt for downloading our final results by running the following cell and skipping the training phases. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 📥 Importing results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!git clone https://github.com/ml6team/quick-tips.git\n",
+    "!cd quick-tips && git checkout feature/long_seq_transformers\n",
+    "!mv quick-tips/nlp/2021_06_29_long_range_transformers/models ./models\n",
+    "!rm -rf quick-tips"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 💪 Training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "id": "PJ6mWoPjVstv"
+   },
+   "outputs": [],
+   "source": [
+    "def compute_metrics(pred):\n",
+    "    labels = pred.label_ids\n",
+    "    preds = pred.predictions.argmax(-1)\n",
+    "    precision, recall, f1, _ = precision_recall_fscore_support(labels, preds, average='binary')\n",
+    "    acc = accuracy_score(labels, preds)\n",
+    "    return {\n",
+    "        'accuracy': acc,\n",
+    "        'f1': f1,\n",
+    "        'precision': precision,\n",
+    "        'recall': recall\n",
+    "    }\n",
+    "\n",
+    "def train_and_evaluate(model, identifier, tokenizer, train_ds, test_ds, max_length=1024):\n",
+    "    def tokenization(batched_text):\n",
+    "        return tokenizer(batched_text['text'], padding='max_length', truncation=True, max_length=max_length)\n",
+    "\n",
+    "    # tokenizing both the training and test dataset \n",
+    "    train_ds = train_ds.map(\n",
+    "        tokenization, batched=True, batch_size=len(train_ds), remove_columns=['text'])\n",
+    "    test_ds = test_ds.map(tokenization, batched=True,\n",
```

Comment:
mapping the tokenization function cuts off the max_length of the testset as well. Do we want to do this for a certain reason?